modules/learn/pages/data/change-history.adoc

```diff
@@ -0,0 +1,245 @@
+The maximum is _1.8 PiB_ (which would be specified as `18446744073709551615`).
+If a positive integer outside this range is specified, an error is flagged, no file-size is established, and change history remains disabled for the bucket.
+
+If the default value of zero is retained for both seconds and bytes, the recording of change history is disabled.
+If a positive integer is specified for either seconds or bytes, the recording of change history is enabled.
+
+Note that the change history is created _per bucket_, and is replicated in accordance with the number of intra-cluster replicas defined for the bucket.
+Therefore, if two replicas are configured, and the specified maximum size is 2 GB, the total size used for the change history across the cluster becomes 6 GB.
+
+== Change-History Management
+
+The following interfaces are provided for configuring change history.
+
+[#change-history-enablement]
+=== Change-History Enablement
+
+Change history can be enabled with either:
+
+* The REST API, using `POST /pools/default/buckets` (to create a bucket) or `POST /pools/default/buckets/<bucketName>` (to edit); specifying one or both of the parameters `historyRetentionBytes` and `historyRetentionSeconds`.
+See xref:rest-api:rest-bucket-create.adoc[Creating and Editing Buckets].
+
+* The CLI, using `couchbase-cli bucket-create` (to create a bucket) or `couchbase-cli bucket-edit` (to edit a bucket); specifying one or both of the parameters `history-retention-bytes` and `history-retention-seconds`.
+See xref:cli:cbcli/couchbase-cli-bucket-create.adoc[bucket-create] and xref:cli:cbcli/couchbase-cli-bucket-edit.adoc[bucket-edit].
+
+[#default-collection-setting]
+=== Default Collection-Setting
+
+If the change history for a bucket is switched _on_, the recording of changes can be performed for collections within the bucket.
+
+A default setting, controlling whether changes are recorded for all collections within the bucket, can be established when a bucket is either created or edited.
+This setting itself has a default value of `true`, which means that change history is indeed recorded for every collection; unless the setting is overridden in the creating or editing of the individual collection.
+
+This bucket-wide default setting can be established with either:
+
+* The REST API, using `POST /pools/default/buckets` (to create a bucket) or `POST /pools/default/buckets/<bucketName>` (to edit); specifying the parameter `historyRetentionCollectionDefault`.
+See xref:rest-api:rest-bucket-create.adoc[Creating and Editing Buckets].
+
+* The CLI, using `couchbase-cli bucket-create` (to create a bucket) or `couchbase-cli bucket-edit` (to edit a bucket); specifying the parameter `enable-history-retention-by-default`.
+See xref:cli:cbcli/couchbase-cli-bucket-create.adoc[bucket-create] and xref:cli:cbcli/couchbase-cli-bucket-edit.adoc[bucket-edit].
+
+[#individual-collection-setting]
+=== Individual Collection-Setting
```

Comment:
I wonder if it's worth a brief hint as to why someone would do this? Overall they can manage the size of their history better if they consider what collections need history and which do not.

Reply:
I've written: This allows a determination to be made as to which collections need their change history to be recorded, and which do not.